inference-engine/src/gna_plugin/layers/gna_convolution_layer.hpp

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <algorithm>
+#include <cmath>
+#include <utility>
+#include <vector>
+
+#include <legacy/ie_layers.h>
+#include "../gna_graph_tools.hpp"
+
+namespace GNAPluginNS {
+struct GNAConvolutionLayer {
+    static bool isMappableFrom2DTo1D(const uint32_t inHeight, const uint32_t inWidth, const uint32_t kernelWidth, const uint32_t strideWidth) {
+        return inHeight > 1 && inWidth > 1 && inWidth == kernelWidth && strideWidth == 1;
+    }
+
+    // 3D input or 2D kernel
+    static bool isConv2D(const uint32_t inHeight, const uint32_t inWidth, const uint32_t inDepth,
+                     const uint32_t kernelHeight, const uint32_t kernelWidth) {
+        return (kernelHeight > 1 && kernelWidth > 1) || (inHeight > 1 && inWidth > 1 && inDepth > 1);
```

Comment:
Should convolution be handled as 2d if inHeight > 1 && inWidth > 1 but inDepth == 1? Currently this case is handled as 2d due to check: ... || in_height != 1. It can be mapped to 1d in some cases but it's done before this check in ConvolutionPrimitive().

Reply:
Currently, Conv2D where 'inDepth == 1' is not supported on GNA HW anyway.
However, you are right `in_height != 1` case is safer to be handled with `finalizeConvolution2DPrimitive()` than `finalizeConvolution1DPrimitive()` (as `finalizeConvolution2DPrimitive()` will throw exception with more friendly description instead of silently hide this buggy condition what finalizeConvolution1DPrimitive() would do).
So, adding `in_height != 1` to dispatch condition in `ConvolutionPrimitive()`.